controller/eks-cluster-config-handler.go

```diff
@@ -1115,15 +1123,19 @@ func (h *Handler) updateUpstreamClusterState(upstreamSpec *eksv1.EKSClusterConfi
 		// in this case update is set right away because creating the
 		// nodegroup may not be immediate
 		if config.Status.Phase != eksConfigUpdatingPhase {
-			config = config.DeepCopy()
 			config.Status.Phase = eksConfigUpdatingPhase
-			var err error
 			config, err = h.eksCC.UpdateStatus(config)
 			if err != nil {
 				return config, err
 			}
 		}
-		ltVersion, err := createNodeGroup(config, ng, eksService, ec2Service, svc)
+		ltVersion, generatedNodeRole, err := createNodeGroup(config, ng, eksService, ec2Service, svc)
+
+		// if a generated node role has not been set on the Status yet and it
+		// was just generated, set it
+		if config.Status.GeneratedNodeRole == "" && generatedNodeRole != "" {
```

Comment:
LGTM, the only potential concerns I might have are that if an error returns later on, and we have to re-reconcile this object, it could potentially create more node groups than necessary (i.e. if another error happens between now and the time we end up updating the config object). I consider that outside of the scope of this PR because it looks like the entirety of this operator (and likely all the others) is written in a way that is contradictory to the design of wrangler's controller framework. We should revisit this at some point to prevent unnecessary thrashing by refactoring this. For example, once `config.Status.GeneratedNodeRole` is set, we should update the config object, and if it fails delete the node group and reenqueue the config object, and if it succeeds, return early since the next reconciliation will shore up the rest of the status updates. Once again, out of the scope of this PR though.